### PR TITLE
Add article about new build specification

### PR DESCRIPTION
**Pull request by astrojuanlu**
As per title.

### CODE REVIEW

File: new-build-specification.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   :author: Juan Luis
+   :location: MAD
+
+.. meta::
+   :description lang=en:
+      You can now use a build image based on Ubuntu 20.04 on Read the Docs
+      with some extra features.
+
+New build specification
+=======================
```
Comment (ericholscher) [inline new-build-specification.rst:12]: This is not a descriptive title. Perhaps `New Ubuntu 20.04 build configurations`?

File: new-build-specification.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+without having to use conda to manage non-Python dependencies.
+
+However, this Ubuntu version was released three years ago and,
+even though it was good enough for most projects,
+we have been receiving requests to upgrade certain system packages over the years.
+Since these are managed through the operating system package manager,
+and we do not provide a way to add custom Personal Package Archives,
+such upgrades were not possible.
+
+On the other hand, many Python projects use a secondary programming language
```
Comment (ericholscher) [inline new-build-specification.rst:36]:
```suggestion
Many Python projects use a secondary programming language
```

File: new-build-specification.rst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+and developers often want specific versions of the toolchains
+that cannot be installed with the system package manager.
+This affected projects that combine backend and frontend code
+that need Node.js to compile their JavaScript sources, such as Jupyter extensions,
+and libraries with performance-critical code written in Rust or Go.
+
+We have often suggested our users to specify custom versions of non-Python dependencies
+`using conda <https://docs.readthedocs.io/en/stable/guides/conda.html>`_,
+to specify custom versions of non-Python dependencies,
+but again this requires learning another tool that might be unfamiliar for some projects.
```
Comment (ericholscher) [inline new-build-specification.rst:49]:
```suggestion
but this requires learning another tool that might be unfamiliar for some projects.
```

File: new-build-specification.rst [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+for a variety of reasons. Established compiled languages like C, C++, and FORTRAN
+have readily available compilers that will work for the majority of use cases.
+However, other younger technologies move at a faster pace,
+and developers often want specific versions of the toolchains
+that cannot be installed with the system package manager.
+This affected projects that combine backend and frontend code
+that need Node.js to compile their JavaScript sources, such as Jupyter extensions,
+and libraries with performance-critical code written in Rust or Go.
+
+We have often suggested our users to specify custom versions of non-Python dependencies
```
Comment (ericholscher) [inline new-build-specification.rst:46]:
```suggestion
We have often suggested specifying custom versions of non-Python dependencies
```

File: new-build-specification.rst [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+`using conda <https://docs.readthedocs.io/en/stable/guides/conda.html>`_,
+to specify custom versions of non-Python dependencies,
+but again this requires learning another tool that might be unfamiliar for some projects.
+
+New ``build.os`` configuration
```
Comment (ericholscher) [inline new-build-specification.rst:51]: This is an odd title, given that it's mostly in the `tools` key? 

```suggestion
New ``build`` YAML configuration
```

File: new-build-specification.rst [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+------------------------------
+
+To overcome all these problems, we have added a new configuration value,
+``build.apt_packages``, that receive a list of APT packages
```
Comment (ericholscher) [inline new-build-specification.rst:55]: Isn't this the old feature, not the new one?

**Review by ericholscher — COMMENTED** [review]
A good start, but I was confused about what we're highlighting in a few places. Guessing just typos/confusion.

File: new-build-specification.rst [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+to specify custom versions of non-Python dependencies,
+but this requires learning another tool that might be unfamiliar for some projects.
+
+New ``build`` YAML configuration
+--------------------------------
+
+To overcome all these problems, we have added a new configuration value,
+``build.tools``, that receive a dictionary of toolchain versions.
+This new configuration requires specifying the base image name
+in ``build.os``, currently `ubuntu-20.04`.
```
Comment (stsewd) [inline new-build-specification.rst:58]:
```suggestion
in ``build.os``, currently ``ubuntu-20.04``.
```

File: new-build-specification.rst [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+   :tags: feature, builders
+   :author: Juan Luis
+   :location: MAD
+
+.. meta::
+   :description lang=en:
+      You can now use a build image based on Ubuntu 20.04 on Read the Docs
+      with some extra features.
+
+Ubuntu 20.04, Python 3.10, and custom versions of Node, Rust, and Go
```
[the remark this answers is not in the document]
Comment (astrojuanlu) [reply new-build-specification.rst:11]: Good point, will change 👍🏽

File: new-build-specification.rst (hunk 7 above)
Comment (astrojuanlu) [reply new-build-specification.rst:58]: Damn, I keep forgetting. Can't I just author everything in MyST now?? 😂

Comment (stsewd) [reply new-build-specification.rst]: Don't go to the darkside!

**Pull request merged**